### DISCUSSION TITLE
Select two unique wildcard draws from two separate calls on the same wildcard

### DISCUSSION THREAD

**Discussion by Sintho** [adieyal/sd-dynamic-prompts#385] [Ideas]
Hey,

I'm currently experimenting a bit and i'm using some wildcards  and i want to have two calls on them
__cloths__:0.6 and  __cloths__:0.4
I know that i can select two distinct wildcards from the same list like this {2$$__cloths__}, but i couldn't find any syntax that would allow me to say
__cloths__=a1:0.6    and  __cloths__!a1:0.4  where i can set a wildcard draw to a variable and use that variable later to say pick a random and if it's the same draw again or did i miss something in the documentation (and would such a thing be even possible in theory)?
And if i'm on the subject would it be possible to add a light script syntax like variables/loops/ifs etc. or is that outside of the scope of an extension for automatica1111?

cheers
Sintho

edit: i jsut saw that my request is similar to the question from https://github.com/adieyal/sd-dynamic-prompts/discussions/375 
but i think a bit different enough warrant it

**Comment by adieyal** [comment]
I don't completely understand your question
What does the syntax `a1` mean in `<lora:__cloths__=a1:0.6>` and `<lora:__cloths__!a1:0.4>`?

Regarding variables, you can create a variable for re-use later with this syntax `${val=!__cloths__}` and then later use it with `${val}`. The standard syntax does not however provide conditionals for you to test whether one variable is the same as another. If you want conditionals and loops, you can try [Jinja2 templates](https://github.com/adieyal/sd-dynamic-prompts/blob/main/jinja2.md. They're a little more complicated to use but can express traditional programming constructs, including loops.

**Reply by Sintho** [reply]
The "a1" in that context was basically your "val=!" ,  simply variable name.
and the "=a1" was setting the result to a variable and "!a1" was excluding the variable from the second selection in pseudocode

From first glace it seems like Jinja2 templates seems to have the script part of what i want, so thanks for the pointer 
I can hopefully accomplish the two distinct wildcard calls without duplicates with thoose added syntax/script options



cheers
Sintho
